README.md

```diff
@@ -21,26 +21,61 @@ This repository provides solutions for configuring [LaunchDarkly code references
+
+A number of command-line arguments are available to the parser, some optional, and some required. Command line arguments may be passed to the program in any order.
+
+### Required arguments
+
+| Option | Description |
+|-|-|
+| `accessToken` | LaunchDarkly [personal access token](https://docs.launchdarkly.com/docs/api-access-tokens) with writer-level access, or access to the `code-references` [custom role](https://docs.launchdarkly.com/v2.0/docs/custom-roles) resource |
+| `dir` | Path to existing checkout of the git repo. If a cloneEndpoint is provided, this option is not required. |
+| `projKey` | A LaunchDarkly project key. |
+| `repoName` | Git repo name. Will be displayed in LaunchDarkly |
 
-## Testing
+Here's an example shell invocation of the git-flag-parser for one of LaunchDarkly's demo repositories:, with the binary located in the current directory, and a minimal configuration:
+
+```shell
+./git-flag-parser \
+  -accessToken="$YOUR_LAUNCHDARKLY_ACCESS_TOKEN" \
+  -dir="path/to/git/repo" \
+  -repoName="my-repository" \
+  -projKey="default"
+```
+
+### Optional arguments
+
+| Option | Description | Default |
+|-|-|-|
+| `baseUri` | Set the base URL of the LaunchDarkly server for this configuration. Only necessary if using a private instance of LaunchDarkly. | `https://app.launchdarkly.com` |
+| `cloneEndpoint` | If provided, will clone the repo from this endpoint to the provided `dir`. If authentication is required, this endpoint should be authenticated. Supports the https protocol for git cloning. Example: `https://username:password@github.com/username/repository.git` | n/a |
+| `contextLines` | The number of context lines to send to LaunchDarkly. If < 0, no source code will be sent to LaunchDarkly. If 0, only the line containing flag references will be sent. If > 0, will send that number of context lines above and below the flag reference. A maximum of 5 context lines may be provided. | -1 |
+| `defaultBranch` | The git default branch. The LaunchDarkly UI will default to display code references for this branch. | "master" |
+| `exclude` | A regular expression defining the files and directories which the flag parser should exclude. | "" |
+| `updateSequenceId` | An integer representing the order number of code reference updates. Used to version updates across concurrent executions of the flag parser. If not provided, data will always be updated. If provided, data will only be updated if the existing `updateSequenceId` is less than the new `updateSequenceId`. Examples: the time a `git push` was initiated, CI build number, the current unix timestamp. | n/a |
+| `repoHead` | The HEAD or ref to retrieve code references from. Should be provided if the `git push` was initiated on a non-master branch. | "master" | no |
```

Comment:
the wording is a bit confusing here. also, can you remind me why we have this option?

Reply:
We have this option so we can decide which branch to checkout to before scanning for code references. I have a separate ticket filed for deciding if we should remove this or not.